test/testenv/ingest_utils.go

```diff
@@ -0,0 +1,207 @@
+		return err
+	}
+
+	// Validate the response of the CLI command
+	var expectedResp strings.Builder
+	fmt.Fprintf(&expectedResp, "Index \"%s\" added.", indexName)
+	if strings.Compare(addIndexResp, expectedResp.String()) == 0 {
+		logf.Log.Error(err, "Failed response to add index to splunk", "pod", podName, "addIndexResp", addIndexResp)
+		return errors.New("Failed response to add index to splunk")
+	}
+
+	logf.Log.Info("Added index to Splunk", "podName", podName, "addIndexResp", addIndexResp)
+	return nil
+}
+
+// IngestFileViaOneshot ingests a file into an instance using the oneshot CLI
+func IngestFileViaOneshot(logFile string, indexName string, podName string, deployment *Deployment) error {
+
+	// Send it to the instance
+	resp, stderr, cpErr := CopyFileToPod(podName, logFile, logFile, deployment)
+	if cpErr != nil {
+		logf.Log.Error(cpErr, "Failed File Copy to pod", "logFile", logFile, "podName", podName, "stderr", stderr)
+		return cpErr
+	}
+	logf.Log.Info("File Copied Successfully", "logFile", logFile, "resp", resp)
+
+	// oneshot log into specified index
+	var addOneshotCmd strings.Builder
+	splunkBin := "/opt/splunk/bin/splunk"
+	username := "admin"
+	password := "$(cat /mnt/splunk-secrets/password) "
+	splunkCmd := "add oneshot"
+
+	fmt.Fprintf(&addOneshotCmd, "%s %s %s -index %s -auth %s:%s", splunkBin, splunkCmd, logFile, indexName, username, password)
+	command := []string{"/bin/bash"}
+	stdin := addOneshotCmd.String()
+	addOneshotResp, stderr, err := deployment.PodExecCommand(podName, command, stdin, false)
+	if err != nil {
+		logf.Log.Error(err, "Failed to execute command on pod", "pod", podName, "stdin", stdin, "addOneshotResp", addOneshotResp, "stderr", stderr)
+		return err
+	}
+
+	// Validate the expected CLI response
+	var expectedResp strings.Builder
+	fmt.Fprintf(&expectedResp, "Oneshot '%s' added", indexName)
+	if strings.Compare(addOneshotResp, expectedResp.String()) == 0 {
+		logf.Log.Error(err, "Failed response to add oneshot to splunk", "pod", podName, "addOneshotResp", addOneshotResp)
+		return err
+	}
+	logf.Log.Info("File Ingested via add oneshot Successfully", "logFile", logFile, "addOneshotResp", addOneshotResp)
+	return nil
+}
+
+// CopyFileToPod copies a file locally from srcPath to the destPath on the pod specified in podName
+func CopyFileToPod(podName string, srcPath string, destPath string, deployment *Deployment) (string, string, error) {
+	// Create tar file stream
+	reader, writer := io.Pipe()
+	if destPath != "/" && strings.HasSuffix(string(destPath[len(destPath)-1]), "/") {
+		destPath = destPath[:len(destPath)-1]
+	}
+	go func() {
+		defer writer.Close()
+		err := cpMakeTar(srcPath, destPath, writer)
+		if err != nil {
+			return
+		}
+	}()
+	var cmdArr []string
+
+	cmdArr = []string{"tar", "-xf", "-"}
+	destDir := path.Dir(destPath)
+	if len(destDir) > 0 {
+		cmdArr = append(cmdArr, "-C", destDir)
+	}
+
+	// Setup exec  command for pod
```

Comment:
Maybe we should stub out whatever is common here and **PodExecCommand** and create a new API. Could be taken up in the future as well.

Reply:
Yes this is the same comment Kriti made.  The testenv code currently doesn't has a separate Http API to handle rest calls so here we are doing it the same as it's done elsewhere in the test code.  It's out of the scope of this PR.  I'll create a backlog item to track it.